src/selectors.ts

```diff
@@ -0,0 +1,27 @@
+import { SelectionEntry, UIDisplayData } from '@aics/simularium-viewer';
+import { CheckboxState, VisibilitySelectionMap } from './constants';
+
+const { Checked } = CheckboxState;
+
+export const getNewHiddenAgents = (
+  uiDisplayData: UIDisplayData,
+  prevCheckboxState: CheckboxState
+): VisibilitySelectionMap => {
+  let payload: VisibilitySelectionMap = {};
+  if (prevCheckboxState === Checked) {
+    payload = uiDisplayData.reduce<VisibilitySelectionMap>((acc, item) => {
```

Comment:
this seems like a default that could be saved and reset?

Reply:
Done, replaced selector with memoized values as we discused